### PR TITLE
Unify `ChainSync` actions under one enum (follow-up)

### PR DESCRIPTION
**Pull request by dmitry-markin**
Get rid of public `ChainSync::..._requests()`  functions and return all requests as actions.

### CODE REVIEW

File: substrate/client/network/sync/src/service/mod.rs [hunk 1]
```diff
@@ -18,6 +18,6 @@
 
 //! `ChainSync`-related service code
 
-pub mod chain_sync;
+pub mod engine;
```
Comment (altonen) [inline substrate/client/network/sync/src/service/mod.rs:21]: Why is `SyncingEngine` moved to service? I don't think this makes sense. I imagined `service` to be for files that pertain to services provide by syncing, such as `NetworkServiceProvider` and `SyncingService`. If there's now tighter coupling between `SyncingService` and `SyncingEngine` and you really want to move files, I think it'd make more sense to have `engine` directory but even that's a push IMO. I thought it was fine as it was before.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4326739

File: substrate/client/network/sync/src/engine.rs [hunk 2]
```diff
@@ -741,7 +738,25 @@ where
...
+				);
+			},
+			ChainSyncAction::SendWarpProofRequest { peer_id, request } => {
+				self.send_warp_proof_request(peer_id, request.clone());
+
+				trace!(
+					target: LOG_TARGET,
+					"Processed `ChainSyncAction::SendWarpProofRequest` to {}, request: {:?}.",
+					peer_id,
+					request,
```
Comment (altonen) [inline substrate/client/network/sync/src/engine.rs:758]:
```suggestion
					"Processed `ChainSyncAction::SendWarpProofRequest` to {peer_id}, request: {request:?}.",
```

**Review by altonen — COMMENTED** [review]
LGTM otherwise but I don't think `service` is the right place for `SyncingEngine`, I thought directory root was a fine place for it

File: substrate/client/network/sync/src/service/mod.rs (hunk 1 above)
Comment (dmitry-markin) [reply substrate/client/network/sync/src/service/mod.rs:21]: It's just the `SyncingService` in the `service` directory. `SyncingEngine` itself is left in the directory root.

File: substrate/client/network/sync/src/engine.rs (hunk 2 above)
Comment (dmitry-markin) [reply substrate/client/network/sync/src/engine.rs:758]: We don't fit here into 100 character per line. It's the only reason why this is not inlined.

**Review by altonen — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

**Review by skunert — APPROVED** [review]

File: substrate/client/network/sync/src/service/engine.rs
Comment (skunert) [inline substrate/client/network/sync/src/service/engine.rs:1]: Hmm maybe `syncing_service.rs` would be better?

**Pull request merged**